content/en/blog/falco-driverkit-debian-docker.md

```diff
@@ -0,0 +1,516 @@
+# once it is compiled inside the container
+
+$ mkdir ./driverkit
+
+# Container with Go tools
+$ docker run --rm -d --name golang-compiler \
+    -v $PWD/driverkit:/export \
+    golang:1.17 sleep infinity
+```
+
+Check that the container has been successfully created and still runs:
+```bash
+$ docker ps
+CONTAINER ID   IMAGE         COMMAND            CREATED         STATUS         PORTS     NAMES
+0a76db23a9d2   golang:1.17   "sleep infinity"   4 seconds ago   Up 3 seconds             golang-compiler
+```
+
+Next, create a shell with a terminal in the container:
+```bash
+$ docker exec -it golang-compiler /bin/bash
+```
+
+Remember, you are in the container context now. Whatever you do here will be lost unless you copy it to the `/export` directory. We will clone the `driverkit` code and compile it using the following commands:
+
+```bash
+git clone https://github.com/falcosecurity/driverkit
+cd driverkit && make
+cp _output/bin/driverkit /export/driverkit
+exit
+```
+
+Once we are done with the Golang container, we can stop it and it'll be automatically removed thanks to the `--rm` parameter that we used to start it.
+
+```bash
+docker stop golang-compiler -t0
+```
+
+#### Creating a configuration file for the driver request
+
+Time to create a configuration file. Do you remember the driver version: `2.0.0+driver`? We will use that and additional information to create the configuration file.
+
+```bash
+# We've included some VARIABLES that will help you understand
+# where the different values come from and what they represent
+
+export DRIVER_VERSION=2.0.0+driver
+export DRIVER_TARGET=debian
+export DRIVER_ARCH=$(arch)
+
+export KERNEL_VERSION=$(uname -v| cut -f1 -d' ' | tr -d \#)
+export KERNEL_RELEASE=$(uname -r)
+export DRIVER_TARGET=debian
+export DRIVER_ARCH=$(arch)
+export DRIVER_NAME=falco
+export PROBE_FILE=${DRIVER_NAME}_${DRIVER_TARGET}_${KERNEL_RELEASE}_${KERNEL_VERSION}.o
+export MODULE_FILE=${DRIVER_NAME}_${DRIVER_TARGET}_${KERNEL_RELEASE}_${KERNEL_VERSION}.ko
+
+mkdir -p drivers/${DRIVER_VERSION}/{${DRIVER_ARCH},}/
+
+# Creating the actual file that we will pass to driverkit
+
+cat > debian.yaml << EOF
+target: ${DRIVER_TARGET}
+driverversion: ${DRIVER_VERSION}
+kernelrelease: ${KERNEL_RELEASE}
+kernelversion: ${KERNEL_VERSION}
+output:
+  module: ./drivers/${DRIVER_VERSION}/x86_64/${MODULE_FILE}
+  probe: ./drivers/${DRIVER_VERSION}/${PROBE_FILE}
+EOF
+```
+
+The resulting file should look like this:
+
+```bash
+$ cat debian.yaml
+target: debian
+driverversion: 2.0.0+driver
+kernelrelease: 5.10.0-14-cloud-amd64
+kernelversion: 1
+output:
+  module: ./drivers/2.0.0+driver/x86_64/falco_debian_5.10.0-14-cloud-amd64_1.ko
+  probe: ./drivers/2.0.0+driver/falco_debian_5.10.0-14-cloud-amd64_1.o
+```
+
+> Pay attention to the different paths since once includes the architecture. They are related to the directory that we have just created and we'll use them later to share it via an HTTP server.
```

Comment:
`Probe` too needs the architecture in the path!

Reply:
Falco 0.32.2 fixed the path for the probe.

Reply:
The blog has been written using Falco 0.32.1, so this means I'll have to update all the outputs (sha256sum, versions, and so on).
This will take a bit more time than just changing just the directory :-)

Reply:
Updated to Falco 0.32.2.